tests/product/image_builder.py

```diff
@@ -0,0 +1,111 @@
+from tests.docker_cluster import DockerCluster
+from tests.hdp_bare_image_provider import HdpBareImageProvider
+from tests.no_hadoop_bare_image_provider import NoHadoopBareImageProvider
+
+from tests.product.base_product_case import BaseProductTestCase
+
+import argparse
+
+
+class ImageBuilder:
+    def __init__(self, caller):
```

Comment:
The name caller suggests that it can be something pretty generic. I think we should do the following:

1. Let this take any testcase (and rename caller -> testcase)
2. Extract the cluster types into tests/product/cluster_types.py and import them here and in base_product_case.

That separates the two dependencies (testcase for assertions and cluster definitions) such that they can be satisfied independently.